### ISSUE TITLE
SingleWorker and horovod

### ISSUE THREAD

**Issue by vlimant** [vlimant/NNLO#21]
the combination --np N --n-processes N does not function while --np 1 --n-processes 1 would

**Comment by abidmalikwaterloo** [comment]
@vlimant Does it mean that it only works with a ring having one GPU and one communicator?

**Comment by vlimant** [comment]
nope. --np 1 --n-processes 1 is making only one process that uses the MPISingleWorker, --np N --n-processes N should actually do something similar, but put the N process in a single ring under N MPISingleWorker ...
